docs/current/guides/757394-use-services.md

```diff
@@ -0,0 +1,347 @@
+
+## Key concepts
+
+Let's start with some important concepts.
+
+Dagger services are just containers, which have the following abilities:
+
+- Each container has its own network namespace and IP address
+- Each container has a unique, deterministic DNS address
+- Containers can expose ports and endpoints
+- Containers can bind other containers as services
+
+Service containers come with the following built-in features:
+
+- Service containers are started just-in-time, deduped, and stopped when no longer needed
+- Service containers are health checked prior to running clients
+- Service containers are given an alias for the client container to use as its hostname
+
+## Using hostnames
+
+Containers run in a bridge network. Each container has its own IP address that other containers can reach. Here's a simple example:
+
+<Tabs groupId="language" className="embeds">
+<TabItem value="Go">
+
+<Embed id="blm2lRNoYzE" />
+
+</TabItem>
+
+<TabItem value="Python">
+
+<Embed id="" />
+
+</TabItem>
+<TabItem value="Node.js">
+
+<Embed id="" />
+
+</TabItem>
+</Tabs>
+
+Containers never use IP addresses to reach each other directly. IP addresses are ephemeral, so doing so would nullify the cache. Instead, Dagger gives each container a unique but deterministic hostname, which doubles as a DNS address. Here's an example:
+
+<Tabs groupId="language" className="embeds">
+<TabItem value="Go">
+
+<Embed id="H5Eb0Hs7JMd" />
+
+</TabItem>
+
+<TabItem value="Python">
+
+<Embed id="" />
+
+</TabItem>
+<TabItem value="Node.js">
+
+<Embed id="" />
+
+</TabItem>
+</Tabs>
+
+This hash value is derived from the same value that determines whether an operation is a cache hit in Buildkit: the vertex digest.
+
+To get a container's address, you wouldn't normally run the `hostname` command, because you'd just be getting the hostname of a container that runs `hostname`, which isn't very helpful. Instead, you would use the hostname API, which returns a domain name reachable by other containers:
+
+<Tabs groupId="language" className="embeds">
+<TabItem value="Go">
+
+<Embed id="cwmeT7388mg" />
+
+</TabItem>
+
+<TabItem value="Python">
+
+<Embed id="" />
+
+</TabItem>
+<TabItem value="Node.js">
+
+<Embed id="" />
+
+</TabItem>
+</Tabs>
+
+In practice, you are more likely to use aliases with `withServiceBinding()`, or `Endpoint()`, which are covered in the next section.
+
+## Exposing ports
+
+Dagger offers the `withExposedPort()` and `Endpoint()` methods to define service ports.
+
+- Use the `withExposedPort()` method to set ports that the container will listen on. Dagger checks the health of each exposed port prior to running any clients that use the service, so that clients don't have to implement their own polling logic.
+- Use the `Endpoint()` method to create a string address to a service's port. You can either specify a port or let Dagger pick the first exposed port.
```

Comment:
nit: capitalization consistency

```suggestion
In practice, you are more likely to use aliases with `withServiceBinding()`, or `endpoint()`, which are covered in the next section.

## Exposing ports

Dagger offers the `withExposedPort()` and `endpoint()` methods to define service ports.

- Use the `withExposedPort()` method to set ports that the container will listen on. Dagger checks the health of each exposed port prior to running any clients that use the service, so that clients don't have to implement their own polling logic.
- Use the `endpoint()` method to create a string address to a service's port. You can either specify a port or let Dagger pick the first exposed port.
```

Reply:
Modified this to reflect the capitalization of the SDK method in each case